src/components/course/data/hooks.js

```diff
@@ -21,6 +31,39 @@ import {
 } from './constants';
 import { features } from '../../../config';
 
+/**
+ * Extracts info needed for enroll button logic from CourseContext
```

Comment:
I appreciate you doing these "docstrings" for JS functions. It's something we could improve on throughout the repo probably 😄 💯